### PR TITLE
[plural entity names] Adding support to plural entity names

### PR DESCRIPTION
**Pull request by lucasncabral**
The ml-git code has been modified to work with `datasets`, `models `and `labels`.

```
ml-git --help

Usage: ml-git [OPTIONS] COMMAND [ARGS]...

Options:
  --version  Show the version and exit.

Commands:
  clone       Clone a ml-git repository ML_GIT_REPOSITORY_URL
  datasets    Management of datasets within this ml-git repository.
  labels      Management of labels sets within this ml-git repository.
  models      Management of models within this ml-git repository.
  repository  Management of this ml-git repository.
```

Logic has also been added to allow automatic updating of an old project. This logic changes the directories of the current project, the keys in the config file and also in the spec file of each entity.

**Example:** 

User workspace:
```
project/
├── dataset
│   └── dataset-ex1
├── labels
│   └── labels-ex1
├── model
│   └── model-ex1
├── .ml-git
│   ├── dataset
│   ├── model
│   ├── label
│   └── config
└── imagenet8.spec
```

Execute any command:
```
> ml-git datasets add dataset-ex
> INFO - MLGit: It was observed that the directories of this project follow the scheme of old versions of ml-git.
>         To continue using this project it is necessary to update the directories.
>         Do you want to update your project now? (Yes/No) Yes
> INFO - MLGit: Project updated successfully
```

After update:
```
project/
├── datasets <- Change here
│   └── dataset-ex1
├── labels
│   └── labels-ex1
├── models <- Change here
│   └── model-ex1
├── .ml-git
│   ├── datasets <- Change here
│   ├── models <- Change here
│   ├── label
│   └── config
└── imagenet8.spec
```

### CODE REVIEW

**Review by Mun1z — APPROVED** [review]

File: ml_git/commands/entity.py [hunk 1]
```diff
@@ -3,14 +3,16 @@
...
 from ml_git.commands.general import mlgit
-from click_didyoumean import DYMGroup
+from ml_git.commands.utils import repositories, LABELS, DATASETS, MODELS
 
 
-@mlgit.group(DATASET, help='Management of datasets within this ml-git repository.', cls=DYMGroup)
-def dataset():
+@mlgit.group(DATASETS, help='Management of datasets within this ml-git repository.', cls=DYMGroup)
+@click.pass_context
+def dataset(ctx):
```
Comment (gilMars) [inline ml_git/commands/entity.py:15]: Could change the function name `def dataset(ctx)` to `def datasets(ctx)`?

File: ml_git/commands/entity.py [hunk 2]
```diff
@@ -25,8 +27,9 @@ def dt_tag_group():
     pass
 
 
-@mlgit.group(MODEL, help='Management of models within this ml-git repository.', cls=DYMGroup)
-def model():
+@mlgit.group(MODELS, help='Management of models within this ml-git repository.', cls=DYMGroup)
+@click.pass_context
+def model(ctx):
```
Comment (gilMars) [inline ml_git/commands/entity.py:32]: Could change the function name `def model(ctx)` to `def models(ctx)`?

File: ml_git/config.py [hunk 3]
```diff
@@ -20,10 +20,10 @@
     'mlgit_path': '.ml-git',
     'mlgit_conf': 'config.yaml',
 
-    'dataset': {
+    'datasets': {
```
Comment (gilMars) [inline ml_git/config.py:23]: Why not use the Enum EntityType?

File: ml_git/config.py [hunk 4]
```diff
@@ -20,10 +20,10 @@
...
         'git': '',
     },
-    'model': {
+    'models': {
```
Comment (gilMars) [inline ml_git/config.py:26]: The same as above.

File: ml_git/config.py [hunk 5]
```diff
@@ -128,8 +128,8 @@ def mlgit_config_save(mlgit_file=__get_conf_filepath()):
         return
 
     config = {
-        'dataset': mlgit_config['dataset'],
-        'model': mlgit_config['model'],
+        'datasets': mlgit_config['datasets'],
+        'models': mlgit_config['models'],
```
Comment (gilMars) [inline ml_git/config.py:132]: Why not use the Enum EntityType?

File: ml_git/config.py [hunk 6]
```diff
@@ -187,25 +187,25 @@ def get_batch_size(config):
...
 
-def get_cache_path(config, type='dataset'):
+def get_cache_path(config, type='datasets'):
     root_path = get_root_path()
     default = os.path.join(root_path, config['mlgit_path'], type, 'cache')
     return getOrElse(config[type], 'cache_path', default)
 
 
-def get_metadata_path(config, type='dataset'):
+def get_metadata_path(config, type=EntityType.DATASETS.value):
```
Comment (gilMars) [inline ml_git/config.py:202]: Could you use the same value for the other functions?

File: ml_git/repository.py [hunk 7]
```diff
@@ -658,7 +654,7 @@ def checkout(self, tag, samples, options):
         options['with_labels'] = False
         if dt_tag is not None:
             try:
-                self.__repo_type = 'dataset'
+                self.__repo_type = 'datasets'
```
Comment (gilMars) [inline ml_git/repository.py:657]: Use the enum EntityType.

File: ml_git/utils.py [hunk 8]
```diff
@@ -69,6 +70,21 @@ def yaml_load(file):
...
     except Exception:
         pass
+    if SPEC_EXTENSION in posix_path(file):
+        hash = check_spec_file(file, hash)
+    return hash
+
+
+def check_spec_file(file, hash):
+    modified = False
+    if 'dataset' in hash:
```
Comment (gilMars) [inline ml_git/utils.py:80]: Assign the value to a variable or use a constant if necessary..

File: ml_git/utils.py [hunk 9]
```diff
@@ -279,3 +295,50 @@ def remove_other_files(filenames, path):
                 os.unlink(file_path)
                 count += 1
     return reclaimed_space, count
+
+
+def check_metadata_directories():
+    try:
+        root_path = get_root_path()
+    except Exception as e:
```
Comment (gilMars) [inline ml_git/utils.py:303]: I guess that you can use the RootPathException:
```
try:
    root_path = get_root_path()
except RootPathException:
    return
```

File: ml_git/utils.py [hunk 10]
```diff
@@ -279,3 +295,50 @@ def remove_other_files(filenames, path):
...
+        if str(e) == 'You are not in an initialized ml-git repository.':
+            return
+        raise e
+
+    have_old_dataset_path = os.path.exists(os.path.join(root_path, 'dataset')) or os.path.exists(os.path.join(root_path, '.ml-git', 'dataset'))
```
Comment (gilMars) [inline ml_git/utils.py:308]: use the constant ROOT_FILE_NAME instead '.ml-git'.

File: ml_git/utils.py [hunk 11]
```diff
@@ -279,3 +295,50 @@ def remove_other_files(filenames, path):
...
+    have_old_model_path = os.path.exists(os.path.join(root_path, 'model')) or os.path.exists(os.path.join(root_path, '.ml-git', 'model'))
+
+    if have_old_dataset_path or have_old_model_path:
+        log.info('It was observed that the directories of this project follow the scheme of old versions of ml-git.\n' +
```
Comment (gilMars) [inline ml_git/utils.py:312]: Transfer the log message to ml_git_message.py if possible.

File: ml_git/commands/utils.py [hunk 12]
```diff
@@ -4,23 +4,24 @@
...
-LABELS = 'labels'
-MODEL = 'model'
+DATASETS = EntityType.DATASETS.value
+LABELS = EntityType.LABELS.value
+MODELS = EntityType.MODELS.value
 PROJECT = 'project'
 
 
-def init_repository(entity_type='dataset'):
+def init_repository(entity_type='datasets'):
```
Comment (GustavoMaciel) [inline ml_git/commands/utils.py:17]:
```suggestion
def init_repository(entity_type=DATASETS):
```
Perhaps it's better to get rid of any hardcoded value as we have the opportunity

File: ml_git/repository.py [hunk 13]
```diff
@@ -811,7 +807,7 @@ def _get_related_tags(self, categories_path, dataset, labels, metadata_path, rep
         dataset_tag, labels_tag = None, None
         spec_path = os.path.join(metadata_path, categories_path, spec_name + '.spec')
         if dataset is True:
-            dataset_tag = get_entity_tag(spec_path, repo_type, 'dataset')
+            dataset_tag = get_entity_tag(spec_path, repo_type, 'datasets')
```
Comment (GustavoMaciel) [inline ml_git/repository.py:810]: Use a constant instead of the hardcoded value here and in the code below

**Review by GustavoMaciel — CHANGES_REQUESTED** [review]
Take a look at the comments I've made and see if they make sense for you as well

**Review by GustavoMaciel — APPROVED** [review]

**Review by gilMars — APPROVED** [review]

File: ml_git/commands/general.py [hunk 14]
```diff
@@ -17,6 +18,7 @@
 @click.version_option(version=get_version(),  message='%(prog)s %(version)s')
 @click.help_option(hidden=True)
 def mlgit():
+    check_metadata_directories()
     pass
```
Comment (diego-pedroso) [inline ml_git/commands/general.py:22]: I believe this line can be removed now.

Comment (lucasncabral) [reply ml_git/commands/general.py]: The line was removed in 427b2f7.

**Review by diego-pedroso — APPROVED** [review]

**Pull request merged**